libraries/AP_Scripting/drivers/TOFSense-M/TOFSense-M_CAN.lua

```diff
@@ -0,0 +1,291 @@
+--[[
+   Driver for NoopLoop TOFSense-M CAN Version. Can be used as a 1-D RangeFidner or 3-D proximity sensor. Upto 3 CAN devices supported in this script although its easy to extend.
+--]]
+
+local update_rate_ms    = 10  -- update rate (in ms) of the driver. 10ms was found to be appropriate
+
+-- Global variables (DO NOT CHANGE)
+local param_num_lua_driver_backend = 36         -- parameter number for lua rangefinder
+local param_num_lua_prx_backend = 15            -- parameter number for lua proximity
+local sensor_setup_done = false
+
+-- Table contains the following info for 3 sensors. If more sensors are needed, this table will need to be increased
+-- approportate scritping backend from rngfnd/prx library, true if backend exists, index parsed last from sensor, minimum distance found since index was 0, Param to decide which rngfnd/prx backednd will match to this sensor, param to decide CAN ID of this sensor 
+local backend_driver = {
+  {lua_driver_backend = nil, sensor_driver_found = false, last_index = 0, min_distance = 0, INSTANCE, CAN_ID},
+  {lua_driver_backend = nil, sensor_driver_found = false, last_index = 0, min_distance = 0, INSTANCE, CAN_ID},
+  {lua_driver_backend = nil, sensor_driver_found = false, last_index = 0, min_distance = 0, INSTANCE, CAN_ID}
+}
+
+local PARAM_TABLE_KEY = 104
+local PARAM_TABLE_PREFIX = "TOFSENSE_"
+
+-- bind a parameter to a variable
+function bind_param(name)
+   local p = Parameter()
+   assert(p:init(name), string.format('could not find %s parameter', name))
+   return p
+end
+
+-- add a parameter and bind it to a variable
+function bind_add_param(name, idx, default_value)
+   assert(param:add_param(PARAM_TABLE_KEY, idx, name, default_value), string.format('could not add param %s', name))
+   return bind_param(PARAM_TABLE_PREFIX .. name)
+end
+
+-- setup parameters
+assert(param:add_table(PARAM_TABLE_KEY, PARAM_TABLE_PREFIX, 15), 'could not add param table')
+
+--[[
+  // @Param: TOFSENSE_PRX
```

Comment:
please check our param parser picks up these parameters

Reply:
![image](https://github.com/ArduPilot/ardupilot/assets/36970042/6911c2e0-3963-4cce-ac53-f5057d391c4a)
It does